calcrule_third_party_payment/calculation_rule.py

```diff
@@ -95,13 +98,22 @@ def check_calculation(cls, instance):
 
     @classmethod
     def calculate(cls, instance, **kwargs):
+        context = kwargs.get('context', None)
         class_name = instance.__class__.__name__
-        # get all “processed“ claims that should be evaluated with fee for service
-        #  that matches args (should replace the batch run)
         if instance.__class__.__name__ == "PaymentPlan":
-            work_data = kwargs.get('work_data', None)
-            cls.convert_batch(work_data=work_data)
-            return "conversion finished 'fee for service'"
+            if context == "BatchPayment":
```

Comment:
```suggestion
           claims = Claim.objects\
                .filter(validity_from__lte=end_date)\
                .filter(validity_from__gte=start_date)\
                .filter(validity_to__isnull=True)\
                .filter(process_stamp__lte=end_date)\
                .filter((Q(items__product=product) | Q(services__product=product)))
            if param['claim_type'] == 'I'
               claims = claims.filter(get_hospital_claim_filter(product.cieling_interpretation))
            elif if param['claim_type'] == 'O'
               claims = claims.filter(~get_hospital_claim_filter(product.cieling_interpretation))
            # no filter required for 'B'
            # same function as for capitation but use the pp param instead of product 
            claims= get_product_hf_filter(product, claims):
                
            if context == "BatchPayment":
```

we need to be more specific regarding the claims, using the get_hospital_claim_filter based on the new "claim_type" param and HF levels